### PR TITLE
etcdctl: add per request timeout

### PR DESCRIPTION
**Pull request by xiang90**
manually tested

10.0.0.1:8212 is unreachable.

```
time ./etcdctl -peers http://10.0.0.1:8212 get foo
client: etcd cluster is unavailable or misconfigured

real	0m1.024s
user	0m0.005s
sys	0m0.006s


time ./etcdctl --timeout 2s --peers http://10.0.0.1:8212 get foo
client: etcd cluster is unavailable or misconfigured

real	0m2.018s
user	0m0.006s
sys	0m0.007s


etcdctl xiangli$ time ./etcdctl --timeout 5s --peers http://10.0.0.1:8212 get foo
client: etcd cluster is unavailable or misconfigured

real	0m5.017s
user	0m0.006s
sys	0m0.008s

```

### CODE REVIEW

File: etcdctl/main.go [hunk 1]
```diff
@@ -37,6 +38,7 @@ func main() {
 		cli.StringFlag{Name: "key-file", Value: "", Usage: "identify HTTPS client using this SSL key file"},
 		cli.StringFlag{Name: "ca-file", Value: "", Usage: "verify certificates of HTTPS-enabled servers using this CA bundle"},
 		cli.StringFlag{Name: "username, u", Value: "", Usage: "provide username[:password] and prompt if password is not supplied."},
+		cli.DurationFlag{Name: "timeout", Value: time.Second, Usage: "connection timeout per request"},
```
Comment (yichengq) [inline etcdctl/main.go]: I would prefer the default value to be longer, like 5s, so it could work well in almost all cases.

Comment (xiang90) [inline etcdctl/main.go]: I would prefer the default to be a more realistic value rather than a long enough one. Or we do not need set the timeout.

**Comment by yichengq** [comment]
One concern is that user may see `client: etcd cluster is unavailable or misconfigured` if his network is slow, and it could be hard to them to find out the reason is too short timeout IMO.

Other than that, LGTM

**Comment by xiang90** [comment]
We have to make the trade off here for the default value. In my opinion, a dead member is much more common than a super slow network (RTT > 1s).  My argument is 1second is not too long at all.

**Comment by yichengq** [comment]
OK.

I think if RTT > 300ms, it could be dangerous. The request may include time from etcdctl to etcd follower, etcd follower to etcd leader and MsgApp/MsgAppResp from/to leader.

The point of my current concern is that considering it may time out, users could be hard to find out the root cause is client timeout IMO. I worry about that `cluster is unavailable or misconfigured` message will let them totally focus on server side and forget about timeout at client side.

**Comment by xiang90** [comment]
@yichengq I think you, now, have another concern. We should print out a better error message. Yes, we can do that, but in another pull request.

**Comment by yichengq** [comment]
I mentioned this in early comment.
LGTM

**Pull request merged**